test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs

```diff
@@ -83,6 +84,123 @@ public void NormalizerWorkout()
             Done();
         }
 
+        [Fact]
+        public void NormalizerParameters()
+        {
+            string dataPath = GetDataPath("iris.txt");
+
+            var loader = new TextLoader(Env, new TextLoader.Arguments
+            {
+                Column = new[] {
+                    new TextLoader.Column("float1", DataKind.R4, 1),
+                    new TextLoader.Column("float4", DataKind.R4, new[]{new TextLoader.Range(1, 4) }),
+                    new TextLoader.Column("double1", DataKind.R8, 1),
+                    new TextLoader.Column("double4", DataKind.R8, new[]{new TextLoader.Range(1, 4) }),
+                    new TextLoader.Column("int1", DataKind.I4, 0),
+                    new TextLoader.Column("float0", DataKind.R4, new[]{ new TextLoader.Range { Min = 1, VariableEnd = true } }),
+                },
+                HasHeader = true
+            }, new MultiFileSource(dataPath));
+
+            var est = new NormalizingEstimator(Env,
+                new NormalizingEstimator.MinMaxColumn("float1"), 
+                new NormalizingEstimator.MinMaxColumn("float4"),
+                new NormalizingEstimator.MinMaxColumn("double1"), 
+                new NormalizingEstimator.MinMaxColumn("double4"),
+                new NormalizingEstimator.BinningColumn("float1", "float1bin"),
+                new NormalizingEstimator.BinningColumn("float4", "float4bin"), 
+                new NormalizingEstimator.BinningColumn("double1", "double1bin"),
+                new NormalizingEstimator.BinningColumn("double4", "double4bin"),
+                new NormalizingEstimator.MeanVarColumn("float1", "float1mv"),
+                new NormalizingEstimator.MeanVarColumn("float4", "float4mv"),
+                new NormalizingEstimator.MeanVarColumn("double1", "double1mv"),
+                new NormalizingEstimator.MeanVarColumn("double4", "double4mv"),
+                new NormalizingEstimator.LogMeanVarColumn("float1", "float1lmv"),
+                new NormalizingEstimator.LogMeanVarColumn("float4", "float4lmv"),
+                new NormalizingEstimator.LogMeanVarColumn("double1", "double1lmv"),
+                new NormalizingEstimator.LogMeanVarColumn("double4", "double4lmv"));
+
+            var data = loader.Read(dataPath);
+
+            var transformer = est.Fit(data);
+
+            var floatAffineData = transformer.ModelParameters["float1"] as NormalizerTransformer.IAffineData<float>;
+            Assert.Equal(0.12658228f, floatAffineData.Scale);
+            Assert.Equal(0, floatAffineData.Offset);
+
+            var floatAffineDataVec = transformer.ModelParameters["float4"] as NormalizerTransformer.IAffineData<ImmutableArray<float>>;
```

Comment:
>NormalizerTransformer.IAffineData<ImmutableArray<float>> [](start = 78, length = 56)

That's the problem that I see with this approach: the `ModelParameters["float4"]` is just a `IModelParameters` (which is as good as to say 'it's an object', because it has no methods).
In reality it is an instance of `NormalizeTransform.AffineColumnFunction.ImplVec<float>`, an internal class that you can't cast to.
But, if you somehow magically know that you need to cast it to `NormalizerTransformer.IAffineData<ImmutableArray<float>>`, you can look at the scales and offsets.

Now what I want is: `transformer.Columns[1]` is a `NormalizerTransformer.ColumnInfo`, and `transformer.Columns[1].ModelParameters` is a `NormalizerModelParametersBase` (which again has no methods).
But if you downcast it to `NormalizerTransformer.AffineVectorModelParameters<float>`, you will get scales and offsets.

The difference is, you know what to downcast to, directly from the debugger. And also in the same debugger you don't have the extra clutter of non-public stuff, because there is no non-public stuff.